File: 프로그래머스/Lv.1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
class Solution {
    public int solution(String[][] board, int h, int w) {
        int minH = 0;
        int maxH = board.length - 1;
        
        int minW = 0;
        int maxW = board[0].length - 1;
        
        int[][] HW = new int[][]{{0,1},{0,-1},{-1,0}, {1,0}};
        
        String targetColor = board[h][w];
        
        int answer = 0;
        
        for(int i = 0;i < 4; i++){
            int searchH = h + HW[i][0];
            int searchW = w + HW[i][1];
            
            if(minH <= searchH && searchH <= maxH 
               && minW <= searchW && searchW <= maxW){
                
                // 같은 색이면
                if(board[searchH][searchW].equals(targetColor))
                    answer++;
            }
            
        }
        
        return answer;
    }
}

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
def solution(board, h, w):
    dy = [0,1,0,-1]
    dx = [1,0,-1,0]
    target = board[h][w]
    cnt = 0
    for i in range(4):
        ny = h + dy[i]
        nx = w + dx[i]
        if 0 <= ny < len(board) and 0 <= nx < len(board[0]) and board[ny][nx] == target: 
            cnt += 1
            
    return cnt